build.sh

```diff
@@ -0,0 +1,239 @@
+#! /bin/bash
+msBuild='/c/Windows/Microsoft.NET/Framework64/v4.0.30319/'
+outputFolder='./_output'
+outputFolderMono='./_output_mono'
+outputFolderOsx='./_output_osx'
+outputFolderOsxApp='./_output_osx_app'
+testPackageFolder='./_tests/'
+testSearchPattern='*.Test/bin/x86/Release/*'
+sourceFolder='./src'
+updateFolder=$outputFolder/NzbDrone.Update
+updateFolderMono=$outputFolderMono/NzbDrone.Update
+
+CheckExitCode()
+{
+    "$@"
+    local status=$?
+    if [ $status -ne 0 ]; then
+        echo "error with $1" >&2
+        exit 1
+    fi
+    return $status
+}
+
+CleanFolder()
+{
+    local path=$1
+    local keepConfigFiles=$2
+
+    echo "Removing XMLDoc files"
+    find $path -name "*.xml" -delete
+    find $path -name "*.transform" -delete
```

Comment:
Maybe that was only added in the new msys with git2... I can switch it to
use -exec rm

On Thu, 23 Jul 2015 20:19 Keivan Beigi <notifications@github.com> wrote:

> In build.sh
> <https://github.com/Sonarr/Sonarr/pull/661#discussion_r35360492>:
>
> > +    local status=$?
> > +    if [ $status -ne 0 ]; then
> > +        echo "error with $1" >&2
> > +        exit 1
> > +    fi
> > +    return $status
> > +}
> > +
> > +CleanFolder()
> > +{
> > +    local path=$1
> > +    local keepConfigFiles=$2
> > +
> > +    echo "Removing XMLDoc files"
> > +    find $path -name "*.xml" -delete
> > +    find $path -name "*.transform" -delete
>
> I'm getting an error here in windows,
>
> Removing XMLDoc files
> find: invalid predicate `-delete'
> find: invalid predicate `-delete'
> find: invalid predicate `-delete'
> Removing FluentValidation.Resources files
> find: invalid predicate `-delete'
> find: invalid predicate `-delete'
> Removing Empty folders
> find: invalid predicate `-delete'
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sonarr/Sonarr/pull/661/files#r35360492>.
>